weather_provider_api/routers/weather/sources/knmi/models/harmonie_arome.py

```diff
@@ -58,63 +57,57 @@ def __init__(self):
+            end: datetime = None,
+            weather_factors: List[str] = None,
     ) -> xr.Dataset:
+        """ Implementation of the WeatherModelBase get_weather function that fetches weather data and returns it as a
+         Xarray DataSet.
+
         """
-            The function that gathers and processes the requested Harmonie Arome weather data from the KNMI site
-            and returns it as an Xarray Dataset.
-            (This model uses the KNMI Data Platform for data acquisition using a repository)
-        Args:
-            coords:             A list of GeoPositions containing the locations the data is requested for.
-            begin:              A datetime containing the start of the period to request data for.
-            end:                A datetime containing the end of the period to request data for.
-            weather_factors:    A list of weather factors to request data for (in string format)
-        Returns:
-            An Xarray Dataset containing the weather data for the requested period, locations and factors.
-        """
-        # Test and account for invalid datetime timeframes or input
-        begin, end = validate_begin_and_end(
-            begin, end, self.repository.get_first_day_of_repo(), self.repository.get_last_day_of_repo()
+        # Handle possibly missing values for this model
+        begin = (begin or datetime.combine(datetime.today(), time.min))  # Fallback: start of today
+        end = (end or datetime.combine(datetime.today() + timedelta(days=1), time.max))  # Fallback: end of tomorrow
+
+        # Validate the given timeframe
+        valid_begin, valid_end = validate_begin_and_end(
+            begin, end, self.repository.first_day_of_repo, self.repository.last_day_of_repo
         )
 
-        ds = self.repository.gather_period(begin, end, coords)
-        ds = ds.sel(
-            prediction_moment=slice(begin, end)
-        )  # Slice of any overflowing time-range
+        # Gather the period from the repository
+        weather_dataset = self.repository.gather_period(begin=valid_begin, end=valid_end, coordinates=coords)
 
-        return ds
+        # Filter the requested factors
+        # TODO: Implement
+        # weather_dataset = ...
+
+        return weather_dataset
 
     def is_async(self):
         return self.async_model
 
-    def _request_weather_factors(self, factors: Optional[List[str]]) -> List[str]:
-        # Implementation of the Base Weather Model function that returns a list of known weather factors for the model.
-        if factors is None:
+    def _request_weather_factors(self, factor_list: Union[List[str], None]) -> List[str]:
+        """ Implementation of the Weather Model Base function that returns a list of the known weather factors out of
+         a given list for this model.
+
+        """
+        if not factor_list:
             return list(self.to_si.keys())
 
-        new_factors = []
+        validated_factors = []
```

Comment:
A very valid argument. The **_request_weather_factors()** function is unfortunately the very definition of code debt. It can easily be changed, but originally it has been added to each and every module, and it has proven to be a pain to transfer to a singular location. 

Changing it here, would mean having to change it for every model. This may happen in the future as a part of the EOL cycle of version 2 of this project, but version 3 currently takes precedence to this.